### DISCUSSION TITLE
_Parallel_choose_min_chunk - typo?

### DISCUSSION THREAD

**Discussion by mikekazakov** [microsoft/STL#2345] [General] [answered]
Hello,
This line looks like a typo:
https://github.com/microsoft/STL/blob/3c2fd04d441d46ec9d914d9cbb621a3bac96c3a5/stl/inc/execution#L1479
`_Static_partitioned_find_end_backward2` should look for the last results, therefore use `_Parallel_choose_max_result`/`_Parallel_choose_max_chunk` like `_Static_partitioned_find_end_forward` does, but for some reason `_Parallel_choose_max_result`/`_Parallel_choose_min_chunk` is used in `_Static_partitioned_find_end_backward2`.
Sorry if I'm missing something.

**Comment by CaseyCarter** [comment, marked as answer]
The difference is that `find_end_backward` uses `_Static_partition_range_backward`, which partitions the range from the end to the beginning - backward - instead of from the beginning to the end. That way chunks with smaller ids are near the end of the range and will be processed first, which enables us to potentially cancel processing of chunks with bigger ids / toward the beginning of the range. (No need to apologize, this is fairly subtle.)

**Reply by mikekazakov** [reply]
Ah, indeed, chunks go backward in that case as well, that is neat!
Thank you very much for the explanation.
